pkg/vex/vex.go

```diff
@@ -169,12 +169,39 @@ func (vexDoc *VEX) ToJSON(w io.Writer) error {
 	return nil
 }
 
+// EffectiveStatement returns the latest VEX statement for a given product, that
+// is the statement that contains the latest data about impact to a given product.
+func (vexDoc *VEX) EffectiveStatement(product, vulnID string) (s *Statement) {
+	for i := range vexDoc.Statements {
+		if vexDoc.Statements[i].Vulnerability == vulnID {
+			isProduct := false
+			for _, p := range vexDoc.Statements[i].Products {
+				if p == product {
+					isProduct = true
+					break
+				}
+			}
+
+			if !isProduct {
+				continue
+			}
+
+			if s == nil {
+				s = &vexDoc.Statements[i]
+			} else if s.Timestamp.Before(*vexDoc.Statements[i].Timestamp) {
+				s = &vexDoc.Statements[i]
```

Comment:
nit: this could be simplified to a single "or" condition since the bodies are the same. But, this may be moot per my other comment on this function.

Reply:
indeed, it's gone